its_hub/algorithms/planning_wrapper.py

```diff
@@ -0,0 +1,303 @@
+        approaches = []
+        
+        # Look for patterns like "APPROACH 1:", "APPROACH 2:", etc.
+        approach_pattern = r'APPROACH\s+(\d+):\s*([^\n]+(?:\n(?!APPROACH)[^\n]*)*)'
+        matches = re.findall(approach_pattern, plan, re.IGNORECASE | re.MULTILINE)
+        
+        for match in matches:
+            approach_num, approach_desc = match
+            # Clean up the approach description
+            approach_desc = approach_desc.strip()
+            approaches.append(approach_desc)
+        
+        # Fallback: if no structured approaches found, try to split by numbered points
+        if not approaches:
+            lines = plan.split('\n')
+            for line in lines:
+                line = line.strip()
+                # Look for numbered approaches like "1.", "2.", "3."
+                if re.match(r'^\d+\.', line):
+                    approach = re.sub(r'^\d+\.\s*', '', line).strip()
+                    if approach:
+                        approaches.append(approach)
+        
+        # Ensure we have at least 2 approaches, fallback to generic ones
+        if len(approaches) < 2:
+            approaches = [
+                "Direct algebraic approach using standard techniques",
+                "Alternative method using different mathematical properties",
+                "Geometric or graphical interpretation approach"
+            ][:max(2, len(approaches))]
+        
+        return approaches[:3]  # Limit to 3 approaches
+
+
+class ApproachPromptTemplate:
+    """Template for generating approach-specific prompts."""
+    
+    APPROACH_TEMPLATE = """Using the {approach} method from your plan, solve this problem step by step:
+
+Problem: {problem}
+
+Approach to use: {approach}
+
+Please solve the problem following this specific approach and show your work clearly. Make sure to box your final answer using \\boxed{{answer}}."""
+
+    @classmethod
+    def create_approach_prompt(cls, problem: str, approach: str) -> str:
+        """Create an approach-specific prompt."""
+        return cls.APPROACH_TEMPLATE.format(problem=problem, approach=approach)
+
+
+class PlanningWrapper(AbstractScalingAlgorithm):
+    """
+    Planning Wrapper that can enhance any ITS algorithm with a planning phase.
+    
+    This wrapper adds a planning step before running the base algorithm, where:
+    1. Model generates a plan with distinct approaches/hypotheses
+    2. Budget is divided equally across planned approaches
+    3. Each approach is executed with the base algorithm
+    4. Best result across all approaches is selected
+    """
+    
+    def __init__(self, base_algorithm: AbstractScalingAlgorithm):
+        """Initialize Planning Wrapper.
+        
+        Args:
+            base_algorithm: The base ITS algorithm to enhance (e.g., SelfConsistency, 
+                           ParticleFiltering, BestOfN, BeamSearch)
+        """
+        self.base_algorithm = base_algorithm
```

Comment:
i think it would be an interesting idea to allow multiple base algs to be allocated across. but this would be a future task.